src/librustc/lint/levels.rs

```diff
@@ -159,19 +123,43 @@ pub struct BuilderPush {
 }
 
 impl<'a> LintLevelsBuilder<'a> {
-    pub fn new(
-        sess: &'a Session,
-        warn_about_weird_lints: bool,
-        sets: LintLevelSets,
-    ) -> LintLevelsBuilder<'a> {
-        assert_eq!(sets.list.len(), 1);
-        LintLevelsBuilder {
+    pub fn new(sess: &'a Session, warn_about_weird_lints: bool, store: &LintStore) -> Self {
+        let mut builder = LintLevelsBuilder {
             sess,
-            sets,
+            sets: LintLevelSets::new(),
             cur: 0,
             id_to_set: Default::default(),
             warn_about_weird_lints,
+        };
+        builder.process_command_line(sess, store);
+        assert_eq!(builder.sets.list.len(), 1);
```

Comment:
Pre-existing, but maybe stick a FIXME here to drop the Vec<> if we're not using it anywhere?

Reply:
Seems like we're using the `Vec<_>` in `fn push`: `self.sets.list.push(LintSet::Node { specs: specs, parent: prev });`.